### ISSUE TITLE
Function.extractParams is deprecated.

### ISSUE THREAD

**Issue by MatthijsBlom** [atom-haskell/ide-haskell#212]
Assigning custom properties to a marker when creating/copying it is
deprecated. Please, consider storing the custom properties you need in
some other object in your package, keyed by the marker's id property.
```
Function.extractParams (<embedded>:127759:20)
MarkerLayer.markPosition (<embedded>:73362:30)
DisplayMarkerLayer.markBufferPosition (<embedded>:128379:58)
TextEditor.addCursorAtBufferPosition (<embedded>:4226:40)
cursors.forEach (C:\Users\Matthijs\.atom\packages\ide-haskell\lib\prettify\index.js:33:24)
Array.forEach (native)
```

**Comment by lierdakil** [comment]
What the hell? What Atom version are you using?

**Comment by lierdakil** [comment]
Quick note: seems TextEditor.addCursorAtBufferPosition(cursor, { autoscroll: false }) triggers this, which is a false positive, since autoscroll is not a marker property, but rather an option for `addCursorAtBufferPosition` specifically.

**Comment by lierdakil** [comment]
I believe https://github.com/atom/atom/commit/b362f746f8fb21d9850f02ca82cc50e90e126d38 is at fault

**Comment by MatthijsBlom** [comment]
> What Atom version are you using?

Atom 1.20, in case you are still wondering.

**Comment by lierdakil** [comment]
Thanks. I'll go ahead and create an issue on atom/atom then.

**Comment by lierdakil** [comment]
Fixed upstream, will be released as Atom 1.22.
